extensions/eclipse-che-theia-plugin-ext/src/browser/che-webview-environment.ts

```diff
@@ -0,0 +1,75 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { CheApiService } from '../common/che-protocol';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+import { WebviewEnvironment } from '@theia/plugin-ext/lib/main/browser/webview/webview-environment';
+import { WebviewExternalEndpoint } from '@theia/plugin-ext/lib/main/common/webview-protocol';
+import { che as cheApi } from '@eclipse-che/api';
+
+@injectable()
+export class CheWebviewEnvironment extends WebviewEnvironment {
+
+    @inject(EnvVariablesServer)
+    protected readonly environments: EnvVariablesServer;
+
+    @inject(CheApiService)
+    private cheApi: CheApiService;
+
+    @postConstruct()
+    protected async init(): Promise<void> {
+        try {
+            const variable = await this.environments.getValue(WebviewExternalEndpoint.pattern);
```

Comment:
do we still need to allow env variable overrides ? as probably it will never work

Reply:
I believe it would be useful for hosted plugin mode.

Reply:
Maybe with some modifications, though

Reply:
ok